app/lib/features/chat/chat_utils/chat_utils.dart

```diff
@@ -51,3 +51,35 @@ UserMentionMessageData parseUserMentionMessage(
     displayName: displayName,
   );
 }
+
+String? getRoomIdFromLink(Uri uri) {
+  // Match regex for matrix room link
+  final urlRegexp = RegExp(
+    r'https://matrix\.to/#/(?<roomId>.+):(?<server>.+)+',
+    caseSensitive: false,
+  );
```

Comment:
`matrix://` is also a valid link format: https://spec.matrix.org/v1.9/appendices/#uris

Reply:
How can I get link like `matrix://` from Acter or Element general?

Reply:
I am not sure which clients are sending those at the moment... the element-x iOS maybe?

Reply:
No, I have check element web and element iOS. Both are sending `matrix.to` url.

And `matrix://` is not a link url,I believe It is `url schema` which is generally used in deep linking.

Reply:
@gnunicorn 
- Is there any platform or way using with I can get `matrix://` and test it?
- If `matrix://` type of url are consider as url schema then still I have to handle condition for it?

Reply:
@kumarpalsinh25 there's a bunch of examples listed in the [spec](https://spec.matrix.org/v1.9/appendices/#matrix-uri-scheme):
![image](https://github.com/acterglobal/a3/assets/40496/164a96f5-3bcf-4697-a705-9eeadd314ff0)

Reply:
matrix.to is the old, `matrix:` is the new way of doing it.

Reply:
> matrix.to is the old, matrix: is the new way of doing it.

@gnunicorn 
Sorry but I'm not agree with that.

Not sure but based on the my knowledge `matrix://` and `matrix.to` are different things and have their own purposes.

- `matrix://` is Matrix URI scheme which ideally suitable in deep-linking for direct intent calls
- `matrix.to` is Matrix Navigation URL and widely used by all the matrix platforms but general navigations.

For better clarification, I'm looking for any platform which is using `matrix://` (Matrix URI Scheme definition). 
If you have any idea about such matrix platforms then can you please share it so that I can look into the same and get detail understanding.

And if you don't mind then can we proceed with this PR as this PR is mainly for copying RoomID URL from profile page and user should able to handle it with click on that message like element and other platforms doing because `matrix://` is different thing and don't have any use-case at least in Acter App.

Reply:
Thanks @gnunicorn for details understanding.
If `matrix://` is the new way then for navigating and deep-link then I would like to integrate that way in Acter and have support for both. 

@gnunicorn Do you have idea of any of the matrix platform which is using this new way?  I just wanted to confirm certain use-cases with that.

Reply:
>  Do you have idea of any of the matrix platform which is using this new way? I just wanted to confirm certain use-cases with that.

as said, if Element-X isn't using it, I wouldn't know if any does. [According to the original spec proposal](https://github.com/matrix-org/matrix-spec-proposals/pull/2312) fluffychat and nheko have support for it (don't know if they post use it primarily though).

FYI, when I say the "new way" of doing it ... the Spec was merged 2021 .... and looks like even Element doesn't use it primarily yet, so ....

Reply:
Aha, I see..
Let me create create new ticket for `"matrix:-protocol-support"` and then we can move on with current `matrix.to` support.